core/pending.go

```diff
@@ -69,6 +147,35 @@ func (p *PreConfirmed) GetTransactionStateDiffs() []*StateDiff {
 	return p.TransactionStateDiffs
 }
 
+func (p *PreConfirmed) GetPreLatest() *PreLatest {
+	return p.PreLatest
+}
+
 func (p *PreConfirmed) Variant() PendingDataVariant {
 	return PreConfirmedBlockVariant
 }
+
+func (p *PreConfirmed) Validate(parent *Header) bool {
```

Comment:
Thinking in abstract here, but shouldn't `PreConfirmed` know who its parent is on creation? How can we create an invalid `PreConfirmed ` or how it's validity could change that we have to call for validation.